### ISSUE TITLE
Texture Not Updating for Large Number of Sprites in Particle Container

### ISSUE THREAD

**Issue by mtderryberry** [pixijs/pixi.js#2518]
Hello,

I am working with Pixi.js and have a problem with updating textures. I have a Particle Container with ~1000 sprites in it. I want to go through the container and change all of their textures to something new. I am currently doing it by the following:

```
function changeTextures() {
    for(var i = 0; i < container.length; i++) {
        var sprite = container.children[i]; 
        container.removeChild(sprite);
        sprite.texture = PIXI.Texture.fromFrame("new_texture"); 
        container.addChild(sprite);
    }
    renderer.render(container);
}
```

This updates most of the sprites, but not all of them. ~100 sprites do not update their textures. However, if I run the function consecutively, more and more sprites will update. Is there a better way to update all sprites in a Containers texture?

**Comment by andrewstart** [comment]
Does using `container.length` let you loop properly? You are reading out of an array that you are changing as you loop through it though - if you don't use `addChild()` here, does it work better?

**Comment by mtderryberry** [comment]
If I do not do addChild(), it does not update at all

**Comment by andrewstart** [comment]
You'll be messing around with a private variable a little, but try setting `container._bufferToUpdate = 0;` after your for loop. In theory that's what should be happening behind the scenes in addChild().

**Comment by englercj** [comment]
Try a reverse loop instead, I think modifying the array in the loop is your problem:

```js
function changeTextures() {
    for(var i = container.children.length - 1; i >= 0; --i) {
        var sprite = container.children[i]; 
        container.removeChild(sprite);
        sprite.texture = PIXI.Texture.fromFrame("new_texture"); 
        container.addChild(sprite);
    }
    renderer.render(container);
}
```

**Comment by mtderryberry** [comment]
I had uvs set to false when creating the Particle Container. Setting it to true would enable me not to have to add/remove the sprite to get it to update?

Also I figured out the problem. I was looping incorrectly. Since addChild puts it at the end of the array, looping incrementally was missing objects. By instead always looking at the first child in the array, I was able to update them all correctly. Thanks for your help!
